oxide-vpc/tests/integration_tests.rs

```diff
@@ -2360,7 +2362,12 @@ fn test_gateway_neighbor_advert_reply(
                 gateway's link-local IPv6 address, generated \
                 from the EUI-64 transform of its MAC",
             );
-            let expected_dst = Ipv6Addr::from_eui64(&g1_cfg.private_mac);
+            let expected_dst = if unicast {
+                Ipv6Addr::from_eui64(&g1_cfg.private_mac)
+            } else {
+                Ipv6Addr::from_eui64(&g1_cfg.private_mac)
+            };
```

Comment:
Something is off here. You switch on `unicast` but return same result. I think you meant for the dst to be either the guest's link-local address or the all-nodes multicast? Which makes it curious why this test isn't failing.

Reply:
Yeah, I was looking at that myself just now. I must have lost some history or forgotten to stash something, since I remember changing that. I'm going to try to reconstruct it, thanks for flagging.